cmd/mdatagen/validate_test.go

```diff
@@ -101,3 +105,38 @@ func TestValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestMetricCollision(t *testing.T) {
+	allMetrics := map[string][]string{}
+	err := filepath.Walk("../../receiver", func(path string, info fs.FileInfo, err error) error {
+		if info.Name() == "metadata.yaml" {
+			// TODO: Remove once https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24011 is merged.
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24011 is merged. Please rebase